homeassistant/components/binary_sensor/maxcube.py

```diff
@@ -0,0 +1,70 @@
+"""
+Support for MAX! Window Shutter via MAX! Cube.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/maxcube/
+"""
+
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.const import STATE_UNKNOWN
+from homeassistant.loader import get_component
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Iterate through all MAX! Devices and add window shutters to HASS."""
+    cube = get_component('maxcube').CUBE_HANDLE.cube
+
+    for device in cube.devices:
+        # Create device name by concatenating room name + device name
+        name = "%s %s" % (cube.room_by_id(device.room_id).name, device.name)
+
+        # Only add Window Shutters
+        if cube.is_windowshutter(device):
+            # add device to HASS
+            add_devices([MaxCubeShutter(name, device.rf_address)], True)
```

Comment:
Please first collect all the devices and then call `add_devices` once with a list of entities to add.

[4.2 on the checklist](https://home-assistant.io/developers/code_review_platform/#4-setup-platform)